doc/src/manual/types.md

```diff
@@ -459,6 +459,27 @@ To recap, two essential properties define immutability in Julia:
       functions as pointers to heap-allocated values except in cases where the compiler
       is sure that there's no way to tell that this is not what is happening.
 
+In cases where one or more fields of an otherwise mutable struct is known to be immutable,
+one can declare these fields as such using `const` as shown below. This enables some,
+but not all of the optimizations of immutable structs, and can be used to enforce invariants
+on the particular fields marked as `const`.
+
+```jldoctest baztype
+julia> mutable struct Baz
+           a::Int
+           const b::Float64
+       end
+
+julia> baz = Baz(1, 1.5);
+
+julia> baz.a = 2
+2
+
+julia> baz.b = 2.0
+ERROR: setfield!: const field .b of type Baz cannot be changed
+[...]
+```
+
```

Comment:
In my opinion, this pull request could be improved by adding a note (following the new material) which says this feature is new in Julia 1.8, similar e.g. to [the compatibility notes in the multi-threading page of the manual](https://github.com/JuliaLang/julia/blob/ee16fe622aabecf7abb0f3e2ec86c98a75ed6241/doc/src/manual/multi-threading.md?plain=1#L25-L30).